types/storybook__addon-knobs/index.d.ts

```diff
@@ -18,35 +18,37 @@ export interface StoryContext {
     story: string;
 }
 
-export interface NumberOptions {
+export type NumberOptions = {
     range: boolean;
     min: number;
     max: number;
     step: number;
-}
+} | {};
```

Comment:
`{}` is a super type of any other type. so this type is basically `{}` which is almost `any`. 

did you mean to make the properties optional instead?

Reply:
e.g.:

```ts
{
     range?: boolean;
     min?: number;
     max?: number;
     step?: number;
}
```

Reply:
also switching from `interface` to `type` is a breaking change to users who can extend this type today. if you do not think users are doing this today, then it should be fine.

Reply:
The `{}` is deliberate, we either need to pass all of them or the default options when omitting them (e.g. an empty object).
Regarding the interface change, I don't think people would extend, but you never know, so I will create an intermediate type.

Reply:
> The {} is deliberate, we either need to pass all of them or the default options when omitting them (e.g. an empty object).

then write it as:
```ts
{
     range: boolean;
     min: number;
     max: number;
     step: number;
} | {
     range?: undefined;
     min?: undefined;
     max?: undefined;
     step?: undefined;
};

```

Reply:
done, ptal